src/model.ts

```diff
@@ -233,7 +233,7 @@ export class Model extends EventBus<any> implements CommandDispatcher {
     this.session.loadInitialMessages(stateUpdateMessages);
     this.joinSession();
 
-    if (config.snapshotRequested) {
+    if (config.snapshotRequested || data["[Content_Types].xml"]) {
```

Comment:
This only works if the user has write access on the record. You also have to check if the model isn't in read-only mode.
I'd target 18.0 because I don't want to spend time fixing more bugs in 16.0 if this introduces an unexpected regression